### ISSUE TITLE
How to set the learning rate of LSTM as 0

### ISSUE THREAD

**Issue by chengzhanzhan** [BVLC/caffe#5174]
Need I revise https://github.com/BVLC/caffe/blob/master/src/caffe/layers/lstm_layer.cpp ?

**Comment by weleen** [comment]
I find the parameters in lstm layer could not be fixed too. Have you solve this problem?

**Comment by weleen** [comment]
@shelhamer I follow the **TODO** instructions  in  [recurrent_layer](https://github.com/BVLC/caffe/blob/master/src/caffe/layers/recurrent_layer.cpp#L281) setting the **layer_need_backward_** to be false, but i find the parameters in LSTM instantiation (e.g. wx_c, b_c, wh_c) is not all fixed, Specifically, wx_c and b_c is fixed and wh_c is getting smaller and smaller with time goes by.
The network is simple. [visualization](http://ethereon.github.io/netscope/#/gist/0f3445d344bb073756dd451e32769c48)
**Iteration 0 (Test)**
![w_xc1](https://user-images.githubusercontent.com/7327734/28854731-77acae66-776b-11e7-8a16-166aedc9060c.png)
![w_hc](https://user-images.githubusercontent.com/7327734/28854734-7e38114e-776b-11e7-80df-3ddf8821f10c.png)
In the training stage, the network should not do backpropagation and the diff is 0.
![data_and_diff](https://user-images.githubusercontent.com/7327734/28855104-555a44c0-776d-11e7-985b-3889a2871b5d.png)
**Iteration 1 (Test)**
![wx_c2](https://user-images.githubusercontent.com/7327734/28854733-7c4bf012-776b-11e7-9ada-99878d1b2b34.png)
![wh_c2](https://user-images.githubusercontent.com/7327734/28854735-7fb3437c-776b-11e7-927a-42f5f7a980d6.png)
and from **lstm_transform_1** to **lstm_transform_16**, the parameters in these layers are not needed backward when the LSTM layer is instantiated.
![lstm_transform](https://user-images.githubusercontent.com/7327734/28854888-1da00d9a-776c-11e7-8709-d93d61ac61ae.png)
